cogs/events.py

```diff
@@ -1,25 +1,64 @@
 import traceback
 import textwrap
 
-from discord import Color, Embed, Guild, Message
+from discord import Color, Embed, Message, Guild, Member
 from discord.ext.commands import Bot, Cog
 
+import re
+import aiohttp
+
 from .utils import constants
 
 PREFIX = constants.COMMAND_PREFIX
 
 
-class Custom(Cog):
+class Events(Cog):
     def __init__(self, bot: Bot) -> None:
         self.bot = bot
         self.dev_mode = constants.DEV_MODE
+        self.session = aiohttp.ClientSession()
+
+    @staticmethod
+    def get_link_code(string: str) -> str:
+        return string.split("/")[-1]
+
+    @staticmethod
+    async def is_our_invite(full_link: str, guild: Guild) -> bool:
+        guild_invites = await guild.invites()
+        for invite in guild_invites:
+            # discord.gg/code resolves to https://discordapp.com/invite/code after using session.get(invite)
+            if Events.get_link_code(invite.url) == Events.get_link_code(full_link):
+                return True
+        return False
 
     @Cog.listener()
     async def on_message(self, message: Message) -> None:
         # Check that the message is not from bot account
         if message.author.bot:
             return
 
+        # DM Check.
+        elif not isinstance(message.author, Member):
+            return
```

Comment:
You can use [`Message.guild`](https://discordpy.readthedocs.io/en/latest/api.html?highlight=message%20guild#discord.Message.guild) to perform a DM check instead of this, that seems much more readable and understandable:
```py
if message.guild is None:
    raise NoPrivateMessage
return True
```